core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlockRequest.kt

```diff
@@ -0,0 +1,36 @@
+package fr.sncf.osrd.api.api_v2.pathfinding
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import fr.sncf.osrd.railjson.schema.rollingstock.RJSLoadingGaugeType
+import fr.sncf.osrd.sim_infra.api.TrackSection
+import fr.sncf.osrd.utils.json.DistanceAdapterFactory
+import fr.sncf.osrd.utils.json.OffsetAdapterFactory
+import fr.sncf.osrd.utils.units.Offset
+
+class PathfindingBlockRequest(
+    @Json(name = "rolling_stock_loading_gauge") val rollingStockLoadingGauge: RJSLoadingGaugeType,
+    @Json(name = "rolling_stock_is_thermal") val rollingStockIsThermal: Boolean,
+    @Json(name = "rolling_stock_supported_electrification")
+    val rollingStockSupportedElectrification: List<String>,
+    @Json(name = "rolling_stock_supported_signaling_systems")
+    val rollingStockSupportedSignalingSystems: List<String>,
+    val timeout: Double?,
+    val infra: String,
```

Comment:
Just so you know, the agreed upon API spec for editoast puts the infra ID [in the URL](https://osrd.fr/en/docs/reference/design-docs/timetable/#path). I'm fine with your version too